File: src/main/java/com/key/oa/config/JacksonConfig.java
package com.key.oa.config;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializerProvider;
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;

import java.io.IOException;

/**
 * @author 孙强
 * 对Jackson进行配置
 */
@Configuration
public class JacksonConfig {
    @Bean
    @Primary
    @ConditionalOnMissingBean(ObjectMapper.class)
    public ObjectMapper jacksonObjectMapper(Jackson2ObjectMapperBuilder builder) {
        ObjectMapper objectMapper = builder.createXmlMapper(false).build();
        // 将Long变成String
        objectMapper.configOverride(Long.class).setFormat(JsonFormat.Value.forShape(JsonFormat.Shape.STRING));
        // 遇到null值时自动转换成空串
        objectMapper.getSerializerProvider().setNullValueSerializer(new JsonSerializer<>() {
            @Override
            public void serialize(Object o, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
                jsonGenerator.writeString("");
            }
        });
        return objectMapper;
    }
}

File: src/main/java/com/key/oa/config/JacksonConfig.kt
package com.key.oa.config

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder

@Configuration
class JacksonConfig {
    @Bean
    @Primary
    @ConditionalOnMissingBean(ObjectMapper::class)
    fun objectMapper(builder: Jackson2ObjectMapperBuilder): ObjectMapper {
        val mapper: ObjectMapper = jacksonObjectMapper()
        // Convert long to string
        mapper.configOverride(Long::class.java).format = JsonFormat.Value.forShape(JsonFormat.Shape.STRING)

        // Convert null object to an empty string
        mapper.serializerProvider.setNullValueSerializer(object : JsonSerializer<Any>() {
            override fun serialize(value: Any?, generator: JsonGenerator, serializers: SerializerProvider) {
                generator.writeString("")
            }
        })
        return mapper
    }
}